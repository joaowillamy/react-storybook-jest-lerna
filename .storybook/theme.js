import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
