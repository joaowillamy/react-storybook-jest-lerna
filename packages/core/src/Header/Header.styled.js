import styled, { css } from "styled-components";
import media from "styled-media-query";

import { chooseScrem } from "../help/chooseScreem";

export const Header = styled.header`
  width: 100vw;
  background-color: #fff;
  padding: 10px;
  box-shadow: ${({ theme }) => theme.shadows.sm()};

  svg {
    width: ${({ theme }) => theme.iconSizes.lg};
    fill: ${({ theme }) => theme.colors.bluePrimary};
  }
`;

export const HeaderNavItens = styled.ul`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const HeaderNavItem = styled.li`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  ${chooseScrem}

  ${media.greaterThan("medium")`
    flex-direction: row;
  `};

  ${({ borderLeftMobile }) => {
    if (borderLeftMobile)
      return css`
        ${media.lessThan("medium")`
              border-left: 2px solid #ccc;
              padding-left: 15px;
         `}
      `;
  }};

  ${({ borderRightMobile }) => {
    if (borderRightMobile)
      return css`
        ${media.lessThan("medium")`
              border-right: 2px solid #ccc;
              padding-right: 15px;
         `}
      `;
  }};

  ${({ borderRightDesktop }) => {
    if (borderRightDesktop)
      return css`
        ${media.greaterThan("medium")`
              border-right: 2px solid #ccc;
              padding-right: 15px;
         `}
      `;
  }};
`;

export const HeaderNavText = styled.p`
  font-size: 12px;
  margin-top: 5px;
  font-weight: 600;

  color: ${({ theme }) => theme.colors.bluePrimary};

  ${media.greaterThan("medium")`
    margin: 0px 10px;
  `};

  ${chooseScrem}
`;

export const HeaderLogo = styled.img`
  width: 120px;
`;

export const HeaderNavFone = styled.span`
  font-size: 14px;
  margin-bottom: 15px;
  font-weight: 800;
`;
