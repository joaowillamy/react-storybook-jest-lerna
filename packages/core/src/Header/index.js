import React from "react";
import PropTypes from "prop-types";
import { Icon } from "../Icons";

import { Container } from "../help/Container.styled";

import * as S from "./Header.styled";

const Header = ({ srcLogo }) => {
  return (
    <S.Header>
      <nav>
        <Container>
          <S.HeaderNavItens>
            <S.HeaderNavItem>
              <ul style={{ display: "inherit" }}>
                <S.HeaderNavItem borderRightMobile borderRightDesktop>
                  <Icon.InfoCircle />
                  <S.HeaderNavText justMobile>Ajuda</S.HeaderNavText>
                  <S.HeaderNavText justDesktop>Como funciona</S.HeaderNavText>
                </S.HeaderNavItem>

                <S.HeaderNavItem justDesktop style={{ marginLeft: "15px" }}>
                  <Icon.Whatsapp style={{ color: "#25d366" }} />
                  <S.HeaderNavText>
                    <S.HeaderNavFone>0800 123 2222</S.HeaderNavFone>
                    <br /> Envie mensagem ou lique
                  </S.HeaderNavText>
                </S.HeaderNavItem>
              </ul>
            </S.HeaderNavItem>

            <S.HeaderNavItem>
              <S.HeaderLogo src={srcLogo} alt={"Logo quero bolsa"} />
            </S.HeaderNavItem>

            <S.HeaderNavItem borderLeftMobile>
              <S.HeaderNavText justDesktop>Nome Sobrenome</S.HeaderNavText>
              <Icon.UserCircle />
              <S.HeaderNavText justMobile>Conta</S.HeaderNavText>
            </S.HeaderNavItem>
          </S.HeaderNavItens>
        </Container>
      </nav>
    </S.Header>
  );
};

Header.propTypes = {
  srcLogo: PropTypes.string
};

Header.defaultProps = {
  srcLogo: "https://via.placeholder.com/885x307"
};

export default Header;
