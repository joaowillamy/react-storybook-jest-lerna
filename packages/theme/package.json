{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
