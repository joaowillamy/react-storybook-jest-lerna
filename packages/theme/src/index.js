export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
