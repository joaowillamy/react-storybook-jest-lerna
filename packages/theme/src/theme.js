const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
