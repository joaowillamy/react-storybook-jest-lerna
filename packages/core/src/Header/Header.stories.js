import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Header from ".";

export default {
  component: Header,
  title: "Core|Header"
};

export const defaultHeader = () => (
  <ThemeProvider>
    <Header>{"Header"}</Header>
  </ThemeProvider>
);
