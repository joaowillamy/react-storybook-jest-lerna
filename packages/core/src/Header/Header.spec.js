import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Header from ".";

describe("Header", () => {
  test("Header", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Header />
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
