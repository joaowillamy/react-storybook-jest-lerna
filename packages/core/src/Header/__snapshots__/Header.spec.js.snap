// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 100vw;
  background-color: #fff;
  padding: 10px;
  box-shadow: 0 1px 3px #00000029;
}

.c0 svg {
  width: 26px;
  fill: #18ACC4;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  font-size: 12px;
  margin-top: 5px;
  font-weight: 600;
  color: #18ACC4;
}

.c7 {
  font-size: 12px;
  margin-top: 5px;
  font-weight: 600;
  color: #18ACC4;
}

.c9 {
  font-size: 12px;
  margin-top: 5px;
  font-weight: 600;
  color: #18ACC4;
}

.c11 {
  width: 120px;
}

.c10 {
  font-size: 14px;
  margin-bottom: 15px;
  font-weight: 800;
}

@media (min-width:768px) {
  .c1 {
    width: 95%;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c4 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (max-width:768px) {
  .c4 {
    border-right: 2px solid #ccc;
    padding-right: 15px;
  }
}

@media (min-width:768px) {
  .c4 {
    border-right: 2px solid #ccc;
    padding-right: 15px;
  }
}

@media (max-width:768px) {
  .c8 {
    display: none;
  }
}

@media (min-width:768px) {
  .c8 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c12 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (max-width:768px) {
  .c12 {
    border-left: 2px solid #ccc;
    padding-left: 15px;
  }
}

@media (min-width:768px) {
  .c6 {
    margin: 0px 10px;
  }
}

@media (min-width:768px) {
  .c6 {
    display: none;
  }
}

@media (min-width:768px) {
  .c7 {
    margin: 0px 10px;
  }
}

@media (max-width:768px) {
  .c7 {
    display: none;
  }
}

@media (min-width:768px) {
  .c9 {
    margin: 0px 10px;
  }
}

<header
  className="c0"
>
  <nav>
    <div
      className="c1"
    >
      <ul
        className="c2"
      >
        <li
          className="c3"
        >
          <ul
            style={
              Object {
                "display": "inherit",
              }
            }
          >
            <li
              className="c4"
            >
              <svg
                aria-hidden="true"
                className="c5"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 16 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
                  fillRule="evenodd"
                />
                <path
                  d="M8.93 6.588l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588z"
                />
                <circle
                  cx={8}
                  cy={4.5}
                  r={1}
                />
              </svg>
              <p
                className="c6"
              >
                Ajuda
              </p>
              <p
                className="c7"
              >
                Como funciona
              </p>
            </li>
            <li
              className="c8"
              style={
                Object {
                  "marginLeft": "15px",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="c5"
                fill="currentColor"
                focusable="false"
                style={
                  Object {
                    "color": "#25d366",
                  }
                }
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M380.9 97.1C339 55.1 283.2 32 223.9 32c-122.4 0-222 99.6-222 222 0 39.1 10.2 77.3 29.6 111L0 480l117.7-30.9c32.4 17.7 68.9 27 106.1 27h.1c122.3 0 224.1-99.6 224.1-222 0-59.3-25.2-115-67.1-157zm-157 341.6c-33.2 0-65.7-8.9-94-25.7l-6.7-4-69.8 18.3L72 359.2l-4.4-7c-18.5-29.4-28.2-63.3-28.2-98.2 0-101.7 82.8-184.5 184.6-184.5 49.3 0 95.6 19.2 130.4 54.1 34.8 34.9 56.2 81.2 56.1 130.5 0 101.8-84.9 184.6-186.6 184.6zm101.2-138.2c-5.5-2.8-32.8-16.2-37.9-18-5.1-1.9-8.8-2.8-12.5 2.8-3.7 5.6-14.3 18-17.6 21.8-3.2 3.7-6.5 4.2-12 1.4-32.6-16.3-54-29.1-75.5-66-5.7-9.8 5.7-9.1 16.3-30.3 1.8-3.7.9-6.9-.5-9.7-1.4-2.8-12.5-30.1-17.1-41.2-4.5-10.8-9.1-9.3-12.5-9.5-3.2-.2-6.9-.2-10.6-.2-3.7 0-9.7 1.4-14.8 6.9-5.1 5.6-19.4 19-19.4 46.3 0 27.3 19.9 53.7 22.6 57.4 2.8 3.7 39.1 59.7 94.8 83.8 35.2 15.2 49 16.5 66.6 13.9 10.7-1.6 32.8-13.4 37.4-26.4 4.6-13 4.6-24.1 3.2-26.4-1.3-2.5-5-3.9-10.5-6.6z"
                  fill="currentColor"
                />
              </svg>
              <p
                className="c9"
              >
                <span
                  className="c10"
                >
                  0800 123 2222
                </span>
                <br />
                 Envie mensagem ou lique
              </p>
            </li>
          </ul>
        </li>
        <li
          className="c3"
        >
          <img
            alt="Logo quero bolsa"
            className="c11"
            src="https://via.placeholder.com/885x307"
          />
        </li>
        <li
          className="c12"
        >
          <p
            className="c7"
          >
            Nome Sobrenome
          </p>
          <svg
            aria-hidden="true"
            className="c5"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8c-1.178 0-2 .822-2 2s.822 2 2 2 2-.822 2-2-.822-2-2-2z"
              fill="none"
            />
            <path
              d="M12 4c-4.337 0-8 3.663-8 8 0 2.176.923 4.182 2.39 5.641A5.007 5.007 0 0111 14.573h2a5.007 5.007 0 014.61 3.068C19.077 16.182 20 14.176 20 12c0-4.337-3.663-8-8-8zm0 10c-2.28 0-4-1.72-4-4s1.72-4 4-4 4 1.72 4 4-1.72 4-4 4z"
              fill="none"
            />
            <path
              d="M13 16.572h-2a3.003 3.003 0 00-2.926 2.354C9.242 19.604 10.584 20 12 20s2.758-.396 3.926-1.073A3.003 3.003 0 0013 16.572z"
              fill="none"
            />
            <path
              d="M12 2C6.579 2 2 6.579 2 12c0 3.189 1.592 6.078 4 7.924V20h.102C7.77 21.245 9.813 22 12 22s4.23-.755 5.898-2H18v-.076c2.408-1.846 4-4.734 4-7.924 0-5.421-4.579-10-10-10zM8.074 18.927A3.002 3.002 0 0111 16.573h2a3.003 3.003 0 012.926 2.354C14.758 19.604 13.416 20 12 20s-2.758-.396-3.926-1.073zm9.536-1.286A5.007 5.007 0 0013 14.573h-2a5.007 5.007 0 00-4.61 3.068C4.923 16.182 4 14.176 4 12c0-4.337 3.663-8 8-8s8 3.663 8 8c0 2.176-.923 4.182-2.39 5.641z"
            />
            <path
              d="M12 6c-2.28 0-4 1.72-4 4s1.72 4 4 4 4-1.72 4-4-1.72-4-4-4zm0 6c-1.178 0-2-.822-2-2s.822-2 2-2 2 .822 2 2-.822 2-2 2z"
            />
          </svg>
          <p
            className="c6"
          >
            Conta
          </p>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
